Domains: Add a8c tags for TLDs still in development

deBhal opened the pull request:
This PR adds an `A8c` tag to the `.ca`, `.de` and `.fr` ccTLDs, to highlight that we're still working on these products and they're not visible to the public yet.

The pre-existing implementation explicitly checked for additional dots (e.g. `example.co.uk` would not match `.uk`), so I've simplified the logic a bit but preserved that behavior.

You can test by clicking the `Domains [Add]` button and entering some domain searches, and confirming that the tag appears only for the three intended tlds (Please don't try and actually register them just yet).

![domain_search_ _julesauspremiumtest_ _wordpress_com](https://cloud.githubusercontent.com/assets/5952255/25511602/12136ba4-2c0b-11e7-8906-8ee14f5e5812.jpg)

yoavf commented:
LGTM, also confirmed the `newTLDs` works as expected when values are added.

klimeryk commented on client/components/domains/domain-registration-suggestion/index.jsx:
```diff
@@ -28,32 +28,42 @@ const DomainRegistrationSuggestion = React.createClass( {
 	render() {
 		const { suggestion, translate } = this.props,
-			isAdded = hasDomainInCart( this.props.cart, suggestion.domain_name ),
+			domain = suggestion.domain_name,
```
Good idea 👍

klimeryk commented on client/components/domains/domain-registration-suggestion/index.jsx:
```diff
@@ -28,32 +28,42 @@ const DomainRegistrationSuggestion = React.createClass( {
+		if ( domain ) {
 			const newTLDs = [];
+			const a8cTLDs = [ '.ca', '.de', '.fr' ];
+			const tld = domain.substring( domain.indexOf( '.' ) );
```
I have mixed feelings about this - while it does nicely fix the issue with the previous approach (not recognizing second level TLDs), it won't work with subdomains... but that should not be an issue, since showing a subdomain here would be pointless. Unless in the future we'll show the subdomain here to suggest it for mapping - but at that point I'm sure we'll need to refactor some other code ;)
Sooo, it's good, but I'm not happy about it 😅 (maybe because I'm jealous that I did not come up with this, but instead the more complicated one 😉 )

aidvu replied:
The `endsWith` was there for TLDs like `co.uk`. But I think this one is ok until we add those.

If we're using `indexOf(.)` we can probably just have the TLDs in the array w/o it.

klimeryk replied:
No, this approach will actually support those better - since we only suggest root domains (not subdomains), everything after the _first_ dot should be a TLD. So, with the previous approach, we'd extract `uk` from `example.co.uk`, but with this one it will be `co.uk` 👍

aidvu replied:
Aaah, nice!

deBhal replied:
The old code handled `co.uk` too, by checking that there's no dot in the "left over" part of the domain [#41](https://github.com/Automattic/wp-calypso/pull/13485/files/bc645b1b5f14bfb1b3c9cfeb7732b40161777881#diff-d94c498e4ad71ff0c5e5bb7fc384bc99L41). I was aiming for simplifying it a little, but preserving the logic.

I share @klimeryk's mixed feelings about subdomains. I think for now I'll add a comment here to make it clearer why we're doing it this way and also mention subdomains.

```node
> suggestion = { domain_name: "example.co.uk" }
{ domain_name: 'example.co.uk' }
> tld = "co.uk"
'co.uk'
> suggestion.domain_name.substring( 0, suggestion.domain_name.length - ( tld.length + 1 ) )
'example'
```

deBhal commented on client/components/domains/domain-registration-suggestion/index.jsx:
```diff
@@ -28,32 +28,42 @@ const DomainRegistrationSuggestion = React.createClass( {
+				domainFlags.push(
+					<DomainSuggestionFlag
+						key={ `${ domain }-a8c` }
+						content={ 'A8c' }
```
Agreed, great suggestion!

deBhal commented on client/components/domains/domain-registration-suggestion/index.jsx:
```diff
@@ -28,32 +28,42 @@ const DomainRegistrationSuggestion = React.createClass( {
+					<DomainSuggestionFlag
+						key={ `${ domain }-a8c` }
+						content={ 'A8c' }
+						status="success"
```
Good idea. Warning seems about right - we don't want people to think twice before selecting it, but they shouldn't feel like they should report it either.

deBhal commented:
Thanks for the suggestions. I've changed the label to a warning "Testing only"
![domain_search_ _my_test_blog_ _wordpress_com](https://cloud.githubusercontent.com/assets/5952255/25786445/88546dca-33d8-11e7-9eb6-8fc7b0d5e6d2.jpg)

I've also added a comment to the tld check, to make it clear that we don't want `.uk` to match `.co.uk` and mention subdomains as a potential problem.

deBhal commented:
I accidentally merged without the feedback improvements but caught that in staging, so I had to merge that change in #13746
